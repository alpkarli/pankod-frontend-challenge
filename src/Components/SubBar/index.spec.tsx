import { create, ReactTestRenderer } from "react-test-renderer";
import { Subbar } from "@Components";

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe("Subbar", () => {
	let component: ReactTestRenderer;
	it("should match snapshot", () => {
		useRouter.mockImplementationOnce(() => ({
			pathname: "/home"
		}))
		component = create(<Subbar />);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it("should change title by router", () => {
		useRouter.mockImplementationOnce(() => ({
			pathname: "/series"
		}))
		component = create(<Subbar />);
		expect(
			component.root.findByType('h2')
				.children
		).toEqual(["Popular Series"]);
	});
	it("should change title to movies by router", () => {
		useRouter.mockImplementationOnce(() => ({
			pathname: "/movies"
		}))
		
		component = create(<Subbar />);

		expect(
			component.root.findByType('h2')
				.children
		).toEqual(["Popular Movies"]);
	});
	it("should work with default", () => {
		useRouter.mockImplementationOnce(() => ({
			pathname: "/entry/a"
		}))
		
		component = create(<Subbar />);

		expect(
			component.root.findByType('h2')
				.children
		).toEqual(["Popular Titles"]);
	});
});
