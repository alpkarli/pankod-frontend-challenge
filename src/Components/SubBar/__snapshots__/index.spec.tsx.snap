// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subbar should match snapshot 1`] = `
<div
  className="sub-header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <h2
        className="sub-header__title"
      >
        Popular Titles
      </h2>
    </div>
  </div>
</div>
`;
